src/Compilers/VisualBasic/Portable/Binding/Binder_Query.vb

```diff
@@ -405,8 +405,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
                 If Not suppressCallDiagnostics AndAlso firstSelectExpression.HasErrors Then
                     Dim query = TryCast(firstSelectExpression, BoundQueryClause)
-
-                    suppressCallDiagnostics = (query.UnderlyingExpression.Kind = BoundKind.BadExpression)
+                    suppressCallDiagnostics = If(query?.UnderlyingExpression, firstSelectExpression).Kind = BoundKind.BadExpression
```

Comment:
@gafter The UnderlyingExpression is never null, firstSelectExpression.Kind was checked before we entered this 'If', all we should check here is whether the query is Nothing. Please adjust the fix by simply putting additional 'If query IsNot Nothing' around original assignment.